docs/content/concepts/assets/asset-checks.mdx

```diff
@@ -0,0 +1,153 @@
+
+```python file=/concepts/assets/asset_checks/asset_with_check.py
+import pandas as pd
+
+from dagster import AssetCheckResult, AssetCheckSpec, Definitions, Output, asset
+
+
+@asset(check_specs=[AssetCheckSpec(name="orders_id_has_no_nulls", asset="orders")])
+def orders(context):
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+
+    # save the output and indicate that it's been saved
+    orders_df.to_csv("orders")
+    yield Output(value=None)
+
+    # check it
+    num_null_order_ids = orders_df["order_id"].isna().sum()
+    yield AssetCheckResult(
+        success=bool(num_null_order_ids == 0),
+    )
+
+
+defs = Definitions(assets=[orders])
+```
+
+### Severity
+
+You can optionally set <PyObject object="AssetCheckSeverity" /> on check results. The default severity is `ERROR`.
+
+```python file=/concepts/assets/asset_checks/severity.py
+from dagster import (
+    AssetCheckResult,
+    AssetCheckSeverity,
+    Definitions,
+    asset,
+    asset_check,
+)
+
+
+@asset
+def my_asset():
+    ...
+
+
+@asset_check(asset=my_asset)
+def my_check():
+    is_serious = ...
+    return AssetCheckResult(
+        success=False,
+        severity=AssetCheckSeverity.ERROR if is_serious else AssetCheckSeverity.WARNING,
+    )
+
+
+defs = Definitions(assets=[my_asset], asset_checks=[my_check])
+```
+
+Severity determines how the check result appears in the UI. If a check fails with `ERROR` severity, the asset will appear red in the lineage graph.
+
+### Adding metadata to check results
+
+You can add information why a check passed or failed using the `metadata` argument on <PyObject object="AssetCheckResult" />. We'll add `num_null_order_ids` as metadata to the `orders_id_has_no_nulls` example:
+
+```python file=/concepts/assets/asset_checks/metadata.py
+import pandas as pd
+
+from dagster import AssetCheckResult, Definitions, asset, asset_check
+
+
+@asset
+def orders():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    orders_df.to_csv("orders.csv")
+
+
+@asset_check(asset=orders, description="Check for null order_ids")
+def orders_id_has_no_nulls():
+    orders_df = pd.read_csv("orders.csv")
+    num_null_order_ids = orders_df["order_id"].isna().sum()
+    return AssetCheckResult(
+        success=bool(num_null_order_ids == 0),
+        metadata={
+            "num_null_order_ids": int(num_null_order_ids),
+        },
+    )
+
+
+defs = Definitions(
+    assets=[orders],
+    asset_checks=[orders_id_has_no_nulls],
+)
+```
+
+There are a variety of types supported via the <PyObject object="MetadataValue" /> class. You can view the metadata on the Checks tab of the asset detail page.
```

Comment:
```suggestion
There are a variety of types supported via the <PyObject object="MetadataValue" /> class. You can view the metadata on the **Checks** tab of the **Asset details** page.
```

UI buttons/labels/page names etc. should be bolded. Page names should also be capitalized.